### ISSUE TITLE
Document that validation of @ConfigurationProperties in a native image requires @Valid to cascade validation

### ISSUE THREAD

**Issue by KimSeongIl** [spring-projects/spring-boot#37101]
spring boot version: 3.1.0
graalvm plugin version: id("org.graalvm.buildtools.native") version "0.9.20"
graalvm version: 22.3.r17

DemoContext.kt
```kotlin
@ConfigurationProperties(prefix = "demo.context")
@Validated
class DemoContext {

    var exclude: List<Exclude> = mutableListOf()

    data class Exclude(
        var url: List<String>? = null,

        var httpMethod: List<HttpMethod>? = null,

        var httpStatus: List<@Pattern(regexp="^([1-5][x|X]{2}|[1-5][0-9]{2})\$") String>? = null,

        var elapsedTime: Long? = null
    )

}
```

application.yml
```yaml
demo:
  context:
    exclude:
      - httpStatus: 2xx
```

nativeRun result
```
Binding to target [Bindable@517a5ddb type = java.util.List<DemoContext$Exclude>, value = 'provided', annotations = array<Annotation>[[empty]]] failed:

    Property: demo.context.exclude[0].httpstatus
    Value: "2xx"
    Origin: class path resource [application.yml] - 31:21
    Reason: The elements [demo.context.exclude[0].httpstatus] were left unbound.

Action:

Update your application's configuration
```

It worked fine before, but when I run it on the native image I get an error.
However, if I remove the Pattern annotation, it works fine.

Why is this problem occurring?
What should I do to resolve this?

**Comment by wilkinsona** [comment]
>  It does not because it has no knowledge of `org.springframework.boot.context.properties.bind.AggregateBinder` logic which is able to analyze `DemoContext` `List<Exclude> property`, to invoke Hibernate validator on individual `Exclude` element

This validation of `Exclude` without `@Valid` is a mistake in the binder that [we've corrected in 3.4](https://github.com/spring-projects/spring-boot/issues/40345).

From Spring Boot's perspective, this should work in a native image if `@Valid` is added to the declaration of the `exclude` field as this should be enough for AOT processing to cascade the search for constraints down into `Exclude` as Hibernate Validator itself does. However, it doesn't work as `BeanValidationBeanRegistrationAotProcessor` doesn't consider cascaded constraints or container element constraints. I've opened https://github.com/spring-projects/spring-framework/issues/33842 so that this can be addressed in Framework.

Once a fix has been made in Framework, we should be able to fix this by documenting in [this section](https://docs.spring.io/spring-boot/docs/3.2.x/reference/html/features.html#features.external-config.typesafe-configuration-properties.validation) that `@Valid` is also needed in a native image.

**Comment by wilkinsona** [comment]
The Framework change has been made in 6.2 only due to its complexity. As such, there's nothing more that we can do here. Boot 3.4 already documents the need to use `@Valid`. In the interim, manual hints will be required to use bean validation with container element constraints and/or constraints that rely on cascading.
